src/Tasks/Common/Resources/Strings.resx

```diff
@@ -758,4 +758,12 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1163: Cannot emit symbols when publishing for .NET 5 with Crossgen2 using composite mode.</value>
     <comment>{StrBegin="NETSDK1163: "}</comment>
   </data>
-</root>
+  <data name="TrimmingWindowsFormsRequiresCom" xml:space="preserve">
+    <value>NETSDK1164: Trimming Windows Forms is dangerous and requires 'BuiltInComSupport' to be set to 'true'.</value>
```

Comment:
I wonder if this should actually be two messages:
- winforms requires com, which is disabled when trimming (hard error to block the scenario initially)
- trimming with BuiltInComSupport true is dangerous (should warn even if the user set it explicitly because some features might not work when trimmed)

Reply:
Is it ok to have that warning for any windows app that sets this property to true and trim? If so, I prefer waiting for the feature switch support but let me know if we should handle it here.

Reply:
I think handling it here would be most correct. Whether BuiltInComSupport is true or false, trimming may break apps that use COM, but at least we can explain this when BuiltInComSupport=false. I'm concerned that setting BuiltInComSupport=true makes errors go away and creates the expectation that things will work, when in reality trimming might still cause problems, as far as I understand. @vitek-karas do you have an opinion on it?

Reply:
But in this case it would be suppressed just because WinForms has `<SuppressTrimAnalysisWarnings>true` right? I don't see why we would make an exception just for COM warnings. It's not like they're any more addressable than the other ones.

Reply:
After some discussion yesterday I think it's ok to rely on the linker warnings (which will be suppressed, but that's a separate concern). My suggestion was based on my understanding that there were COM activation scenarios that the linker wouldn't be able to detect, but it sounds like that's not true (aside from some possible edge cases).